### ISSUE TITLE
Metaclasses handled incorrectly

### ISSUE THREAD

**Issue by danemtsov** [microsoft/pyright#920]
Valid calls to metaclass methods don't pass type checking. Minimal example:

```python
class Meta(type):
    def foo(cls, some_param):
        print(cls, "foo called with", some_param)

class Clazz(metaclass=Meta):
    pass

Clazz.foo("some argument") # prints "<class '__main__.Clazz'> foo called with some argument"
    # Error messages:
    # (1) Argument missing for parameter "some_param"
    # (2) Argument of type "Literal['some argument']" cannot be assigned to parameter "cls" of
    #     type "Meta" in function "foo"
    #       "Literal['some argument']" is incompatible with "Meta"
```

In contrast
```python
Clazz.foo(Meta("X", (), {}), "some argument")
```
passes type checking but is wrong use.

Pyright seems not to recognize metaclass methods (here `foo(...)`) as methods of the deriving class object (here `<class 'Clazz'>`).
It seems to treat them as static methods of the deriving class, wanting explicit arguments for every parameter instead of using `<class 'Clazz'>` as an implicit argument for `cls`.

A more common case for this is e.g. `abc.ABCMeta.register(...)`.

---

Running Pyright as part of the Pylance VS Code extension:
Id: ms-python.vscode-pylance
Version: 2020.8.0
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of Pyright and Pylance.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.61, which I just published. It will also be in the next release of Pylance.
